### PR TITLE
Add tool for bulk updating references to the AWS Rust SDK

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
This is needed for running examples from the [aws-doc-sdk-examples](https://github.com/awsdocs/aws-doc-sdk-examples) repository against the generated SDK in `smithy-rs`'s CI.

## Testing
- Manually tested against `aws-doc-sdk-examples` locally

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/sdk-versioner/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  --sdk-version 0.5.0 \
+  --smithy-version 0.35.0 \
+  /path/to/aws-doc-sdk-examples/rust_dev_preview
+```
+
+Example updating SDK examples to refer to local generated code:
+```bash
+$ sdk-versioner \
+  --sdk-path /path/to/smithy-rs/aws/sdk/build/aws-sdk/sdk \
+  /path/to/aws-doc-sdk-examples/rust_dev_preview
```
Comment (Velfi) [inline tools/sdk-versioner/README.md:22]: Does the tool work with relative paths as well?

File: tools/sdk-versioner/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+#[derive(StructOpt, Debug)]
+#[structopt(
+    name = "sdk-versioner",
+    about = "CLI tool to recursively update SDK/Smithy crate references in Cargo.toml files"
+)]
+struct Opt {
+    /// Path(s) to recursively update Cargo.toml files in
+    #[structopt()]
+    crate_paths: Vec<String>,
```
Comment (Velfi) [inline tools/sdk-versioner/src/main.rs:33]: What about typing them as paths too?
```suggestion
    crate_paths: Vec<PathBuf>,
```

File: tools/sdk-versioner/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+            [dependencies]
+            aws-config = { version = "0.5.0" }
+            aws-sdk-s3 = { version = "0.5.0" }
+            aws-smithy-types = { version = "0.35.0" }
+            aws-smithy-http = { version = "0.35.0", features = ["test-util"] }
+            something-else = "0.1"
+            "#,
+        );
+    }
```
Comment (Velfi) [inline tools/sdk-versioner/src/main.rs:239]: testing with inline `Cargo.toml`s is good but it's probably worth adding a filesystem-based test with [tempfile](https://docs.rs/tempfile/latest/tempfile/) like I did for the [sync tool fs operations](https://github.com/awslabs/smithy-rs/blob/21ffa905dbc048dde6dff654bf953ee01df478b5/tools/smithy-rs-sync/src/fs.rs#L217)

File: tools/sdk-versioner/README.md (hunk 1 above)
Comment (jdisanti) [reply tools/sdk-versioner/README.md:22]: It does. Removed the leading slash to make this more clear.

**Review by Velfi — APPROVED** [review]

**Pull request merged**